pulser/_seq_drawer.py

```diff
@@ -138,6 +143,16 @@ def phase_str(phi):
         ya = data[ch]['amp']
         yb = data[ch]['detuning']
 
+        if len(time_slices) > 0:
+            teff = []
+            yaeff = []
+            ybeff = []
+
+            for ti, tf in zip(time_slices, time_slices[1:]):
+                teff += [t[ti], t[tf]]
+                yaeff += [ya[ti], ya[ti]]
+                ybeff += [yb[ti], yb[ti]]
+
```

Comment:
I'm a bit confused about this. Doesn't this work?
```suggestion
            teff = t[time_slices]
            yaeff = np.array(ya)[time_slices]
            ybeff = np.array(yb)[time_slices]
```
Perhaps you can even convert `ya` and `yb` into `np.arrays` when they're first defined, not sure without trying.

Reply:
There was indeed an issue there, that the test discovered : if the time steps in t are not regularly spaced, it wouldn't work. I had to change it and it now seems to be working fine on my machine

Reply:
That's were you get me, are you sure that it's a "constant-by-part" shape we're doing? Is it not piecewise-linear instead?

Reply:
That's the impression I had after discussing with Sebastian, I'll check again, to be sure

Reply:
I've found the answer to that question in the `QobjEvo` documentation: 

> By default, a cubic spline interpolation will be used for the coefficient 
> at time t. If the coefficients are to be treated as step functions, use the arguments 'args = {"_step_func_coeff": True}'.

In the end, I think the "effective" pulse will always be close to the target one, since the `sampling_rate` only affect measurements, not the time evolution itself

Reply:
The question I have then is how our device behaves within this 4ns time window. If the user wants to compare

1. The pulse they input
2. The pulse the device would be _actually_ creating
3. The pulse Qutip is used in the simulation

Reply:
Even though the duration of the pulses has to be a multiple of 4ns, the device can define the amplitude at each 1ns. I would say that, at this level, it is irrelevant whether it changes in steps or ramps between each ns time step. For the detuning, it is defined as piecewise-linear ramps of 4ns. 

In any case, what happens on the device is not what we're looking for here, is it? This is supposed to be a realistic representation of the input to the simulation, not the device controller's output.

Reply:
Bear in mind that this is not terribly relevant. We are discretely modulating the amplitude of a wave of frequency in the tens of MHz with a sampling rate of 1 GHz. For the controller, all that really matters is the amplitude value at each time step, not what's in between.

Reply:
I agree. I have a strange issue raised by pytest when testing this using sthe spline interpolation :
apparently in one of the pulses used in the test, `np.array(data[ch]['time'])` is not an increasing array - namely there are three points like this one `3999.0 4000.0 3999.0` around 4000, 5000, 6000 and 7000.
Is it coming from the addition of pulses in the sequence creation?